internal/backends/qt/qt_window.rs

```diff
@@ -1342,6 +1020,172 @@ impl ItemRenderer for QtItemRenderer<'_> {
+            return;
+        };
+
+        raw_font.set_pixel_size(font_size);
+
+        let (glyph_indices, positions): (Vec<u32>, Vec<qttypes::QPointF>) = glyphs_it
+            .into_iter()
+            .map(|g| {
+                (g.id as u32, qttypes::QPointF { x: g.x as f64, y: g.y as f64 + y_offset as f64 })
+            })
+            .unzip();
+
+        let glyph_indices_ptr = glyph_indices.as_ptr();
+        let glyph_positions_ptr = positions.as_ptr();
+        let size: u32 = glyph_indices.len() as u32;
+
+        let mut qt_pen = qttypes::QPen::default();
+        let mut qt_brush = qttypes::QBrush::default();
+
+        let painter: &mut QPainterPtr = &mut self.painter;
+
+        match brush {
+            GlyphBrush::Fill(qbrush) => qt_brush = qbrush,
+            GlyphBrush::Stroke(qpen) => qt_pen = qpen,
+        }
+
+        cpp! { unsafe [painter as "QPainterPtr*", glyph_indices_ptr as "const quint32 *", glyph_positions_ptr as "const QPointF *", size as "int", qt_pen as "QPen", qt_brush as "QBrush", raw_font as "QRawFont"] {
+            if (size == 0)
+                return;
+            (*painter)->setPen(qt_pen);
+            (*painter)->setBrush(qt_brush);
+            QGlyphRun glyphRun;
+            glyphRun.setRawFont(raw_font);
+            glyphRun.setRawData(glyph_indices_ptr, glyph_positions_ptr, size);
+            (*painter)->drawGlyphRun(QPointF(0, 0), glyphRun);
+        }}
+    }
+
+    fn fill_rectangle(
+        &mut self,
+        physical_x: f32,
+        physical_y: f32,
+        physical_width: f32,
+        physical_height: f32,
+        color: i_slint_core::Color,
+    ) {
+        let rect = qttypes::QRectF {
+            x: physical_x as _,
+            y: physical_y as _,
+            width: physical_width as _,
+            height: physical_height as _,
+        };
+        let color: u32 = color.as_argb_encoded();
+        let painter: &mut QPainterPtr = &mut self.painter;
+        cpp! { unsafe [painter as "QPainterPtr*", color as "QRgb", rect as "QRectF"] {
+            (*painter)->fillRect(rect, QBrush(QColor::fromRgba(color)));
+        }}
+    }
+}
+
+cpp_class! {pub unsafe struct QRawFont as "QRawFont"}
+
+impl QRawFont {
+    pub fn load_from_data(&mut self, data: &[u8], pixel_size: f32) {
+        let font_data = qttypes::QByteArray::from(data);
+        cpp! { unsafe [ self as "QRawFont*", font_data as "QByteArray", pixel_size as "float"] {
+            self->loadFromData(font_data, pixel_size, QFont::PreferDefaultHinting);
+        }}
+    }
+
+    pub fn set_pixel_size(&mut self, pixel_size: f32) {
+        cpp! { unsafe [ self as "QRawFont*", pixel_size as "float"] {
+            self->setPixelSize(pixel_size);
+        }}
+    }
+
+    pub fn is_valid(&self) -> bool {
+        cpp! { unsafe [ self as "const QRawFont*"] -> bool as "bool" {
+            return self->isValid();
+        }}
+    }
+}
+
+pub struct FontCache {
+    fonts: HashMap<(u64, u32), Option<QRawFont>>,
```

Comment:
a comment explaining what the key of this cache is would be nice.

Is there a risk that the cache grows unbounded? The font size is not part of it, right?

Reply:
I'll add a comment. The cache grows with the number of fonts used in the application (so types, no instances). I'll make a separate change for a more experimental approach :)